app/models/manageiq/providers/vmware/infra_manager.rb

```diff
@@ -649,6 +649,14 @@ def vm_quick_stats(obj, options = {})
     end
     alias_method :host_quick_stats, :vm_quick_stats
 
+    def vm_set_description(vm, new_description, options = {})
+      options[:spec] = VimHash.new("VirtualMachineConfigSpec") do |spec|
+        spec.annotation = new_description
+      end
+
+      vm_reconfigure(vm, options)
```

Comment:
Changing the annotation is a `ReconfigVM_Task` operation so we can reuse the existing `vm_reconfigure` method